components/style/values/specified/position.rs

```diff
@@ -608,6 +610,37 @@ impl Parse for TemplateAreas {
 
 trivial_to_computed_value!(TemplateAreas);
 
+#[derive(Clone, Debug, PartialEq, ToCss)]
+/// Arc type for `Arc<TemplateAreas>`
+pub struct TemplateAreasArc(pub Arc<TemplateAreas>);
+
+impl Parse for TemplateAreasArc {
+    fn parse<'i, 't>(
+        _context: &ParserContext,
+        input: &mut Parser<'i, 't>,
+    ) -> Result<Self, ParseError<'i>> {
+        let mut strings = vec![];
+        while let Ok(string) = input.try(|i| i.expect_string().map(|s| s.as_ref().into())) {
+            strings.push(string);
+        }
+
+        match TemplateAreas::from_vec(strings) {
+            Ok(template_areas) => Ok(TemplateAreasArc(Arc::new(template_areas))),
+            Err(_) => Err(input.new_custom_error(StyleParseErrorKind::UnspecifiedError))
+        }
+    }
+}
+
+impl MallocSizeOf for TemplateAreasArc {
+    /// FIXME: (cybai) NEED TO FIGURE OUT HOW TO IMPLEMENT
+    /// MallocConditionalSizeOf for this Arc
+    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+        0
```

Comment:
@nnethercote, what's the best way to measure this? For now measuring as `0` may be fine, since this is pretty rare, but in case it comes up again...